src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -292,20 +303,32 @@ private Channel createChannel() {
     }
 
     private void waitForConnectionRecovery(ShutdownNotifier notifier) {
-        while (!notifier.isOpen() && !connectionIsClosed.get()) {
+        boolean isClose = isConnectionRecovery(notifier);
+
+        if (isClose) {
+            LOGGER.warn("Start wait for recovery connection");
+        }
+
+        while (isClose) {
```

Comment:
In my opinion, splitting the condition and its recalculation might cause some problems.
I would either call `isConnectionRecovery` in the `while` condition or convert the `while` loop to `do-while`. Like this:
```java
do {
...
} while(isConnectionRecovery(notifier))
```